src/libstd/panicking.rs

```diff
@@ -258,16 +258,25 @@ pub mod panic_count {
         LOCAL_PANIC_COUNT.with(|c| c.get())
     }
 
+    #[inline]
     pub fn is_zero() -> bool {
         if GLOBAL_PANIC_COUNT.load(Ordering::Relaxed) == 0 {
             // Fast path: if `GLOBAL_PANIC_COUNT` is zero, all threads
             // (including the current one) will have `LOCAL_PANIC_COUNT`
             // equal to zero, so TLS access can be avoided.
             true
         } else {
-            LOCAL_PANIC_COUNT.with(|c| c.get() == 0)
+            is_zero_slow_path()
         }
     }
+
+    // Slow path is in a separate function to reduce the amount of code
+    // inlined from `is_zero`.
+    #[inline(never)]
```

Comment:
`#[cold]` should be enough, as a general rule we avoid using both cold and `inline(never)`.

Reply:
I did that way because that is how it is done here:
https://github.com/rust-lang/rust/blob/0b66a89735305ebac93894461559576495ab920e/src/libcore/option.rs#L1262-L1267

Reply:
Fair enough, I guess it doesn't really matter in practice.